### PR TITLE
Don't replicate the original stmt when doing specialize_fail()

### PR DESCRIPTION
**Pull request by shoaibkamil**
Proposed fix for #3926.

### CODE REVIEW

File: src/ScheduleFunctions.cpp [hunk 1]
```diff
@@ -371,8 +373,11 @@ Stmt build_provide_loop_nest(const map<string, Function> &env,
                                      {StringImm::make(s.failure_message)},
                                      Internal::Call::Extern);
             then_case = AssertStmt::make(const_false(), specialize_fail_error);
+            // Since the condition is a constant true, the else case will
+            // never execute
+            else_case = then_case;
         }
-        stmt = IfThenElse::make(s.condition, then_case, stmt);
+        stmt = IfThenElse::make(s.condition, then_case, else_case);
```
Comment (abadams) [inline src/ScheduleFunctions.cpp]: Isn't this constructing IfThenElse::make(s.condition, then_case, then_case)? Why not just set stmt = then_case in the else branch above and only construct the IfThenElse in the empty-failure-message branch.

Comment (shoaibkamil) [reply src/ScheduleFunctions.cpp]: Very true.  Pushed a simpler version that doesn't construct a redundant IfThenElse.

**Comment by shoaibkamil** [comment]
Failures seem to be `simd_op_check`.  Ok to merge @steven-johnson?

File: test/correctness/issue_3926.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+int main(int argc, char* argv[]) {
+  Func f("f"), g("g");
+  Var x("x"), y("y");
+  Var tx("tx"), ty("ty");
+  Param<bool> param;
+
+  f(x) = x;
+  g(x, y) = f(x) + select(param, 1, 2);
+
+  //g.gpu_tile(x, y, tx, ty, 8, 8, TailStrategy::GuardWithIf);
```
Comment (steven-johnson) [inline test/correctness/issue_3926.cpp]: Just noticed this, but: did you mean to leave this line commented out?

**Pull request merged**